netaddr.go

```diff
@@ -29,50 +29,40 @@ const (
 	maxUint16 = 1<<16 - 1
 )
 
+// o4 is the offset to the beginning of the IPv4 address within IP.a.
+const o4 = 12
+
 // Sizes: (64-bit)
 //   net.IP:     24 byte slice header + {4, 16} = 28 to 40 bytes
 //   net.IPAddr: 40 byte slice header + {4, 16} = 44 to 56 bytes + zone length
-//   netaddr.IP: 16 byte interface + {4, 16, 24} = 20, 32, 40 bytes + zone length
+//   netaddr.IP: 24 bytes, always (zone, if any, is cached globally forever)
 
 // IP represents an IPv4 or IPv6 address (with or without a scoped
 // addressing zone), similar to Go's net.IP or net.IPAddr.
 //
 // Unlike net.IP or net.IPAddr, the netaddr.IP is a comparable value
 // type (it supports == and can be a map key) and is immutable.
-// Its memory representation ranges from 20 to 40 bytes, depending on
-// whether the underlying adddress is IPv4, IPv6, or IPv6 with a
-// zone. (This is smaller than the standard library's 28 to 56 bytes)
+// Its memory representation is 24 bytes in 64-bit machines (the same
```

Comment:
It's going to be 24 bytes on all machines, right? There are no more machine-dependent types in there.